pkg/resources/object_renaming_lists_and_sets_acceptance_test.go

```diff
@@ -0,0 +1,463 @@
+					}),
+				),
+			},
+			// Remove, shift, and add one item (at the beginning)
+			{
+				Config: objectRenamingConfigList([]map[string]any{
+					{"name": "", "string": "222", "int": 222},
+					{"name": "", "string": "333", "int": 333},
+					{"name": "", "string": "444", "int": 444},
+				}),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					assert.HasListItemsOrderIndependent("snowflake_object_renaming.test", "list", []map[string]string{
+						{"name": "", "string": "222", "int": "222"},
+						{"name": "", "string": "333", "int": "333"},
+						{"name": "", "string": "444", "int": "444"},
+					}),
+				),
+			},
+			// Reorder items and add one
+			{
+				Config: objectRenamingConfigList([]map[string]any{
+					{"name": "", "string": "444", "int": 444},
+					{"name": "", "string": "555", "int": 555},
+					{"name": "", "string": "333", "int": 333},
+					{"name": "", "string": "222", "int": 222},
+				}),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					assert.HasListItemsOrderIndependent("snowflake_object_renaming.test", "list", []map[string]string{
+						{"name": "", "string": "444", "int": "444"},
+						{"name": "", "string": "555", "int": "555"},
+						{"name": "", "string": "333", "int": "333"},
+						{"name": "", "string": "222", "int": "222"},
+					}),
+				),
+			},
+			// Replace all items
+			{
+				Config: objectRenamingConfigList([]map[string]any{
+					{"name": "", "string": "444", "int": 444},
+					{"name": "", "string": "555", "int": 555},
+					{"name": "", "string": "333", "int": 333},
+					{"name": "", "string": "222", "int": 222},
+				}),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					assert.HasListItemsOrderIndependent("snowflake_object_renaming.test", "list", []map[string]string{
+						{"name": "", "string": "444", "int": "444"},
+						{"name": "", "string": "555", "int": "555"},
+						{"name": "", "string": "333", "int": "333"},
+						{"name": "", "string": "222", "int": "222"},
+					}),
+				),
+			},
+			// Add an item that is identical to another one (currently, failing)
+			// {
+			//	Config: objectRenamingConfigList([]map[string]any{
+			//		{"name": "", "string": "444", "int": 444},
+			//		{"name": "", "string": "555", "int": 555},
+			//		{"name": "", "string": "333", "int": 333},
+			//		{"name": "", "string": "222", "int": 222},
+			//		{"name": "", "string": "222", "int": 222},
+			//	}),
+			//	Check: resource.ComposeAggregateTestCheckFunc(
+			//		assert.HasListItemsOrderIndependent("snowflake_object_renaming.test", "list", []map[string]string{
+			//			{"name": "", "string": "444", "int": "444"},
+			//			{"name": "", "string": "555", "int": "555"},
+			//			{"name": "", "string": "333", "int": "333"},
+			//			{"name": "", "string": "222", "int": "222"},
+			//			{"name": "", "string": "222", "int": "222"},
+			//		}),
+			//	),
+			// },
+		},
+	})
+}
+
+// This test researches the possibility of performing update instead of remove + add item
```

Comment:
how this test proves/disproves this possibility if we only check the final state?